state-sync/storage-service/server/src/lib.rs

```diff
@@ -364,18 +392,24 @@ impl StorageReaderInterface for StorageReader {
         &self,
         _version: u64,
         _start_account_index: u64,
-        _expected_num_account_states: u64,
+        _end_account_index: u64,
     ) -> Result<AccountStatesChunkWithProof, Error> {
         // TODO(joshlind): implement this once DbReaderWriter supports these calls.
         Err(Error::UnexpectedErrorEncountered(
             "Unimplemented! This API call needs to be implemented!".into(),
         ))
     }
+}
 
-    fn get_number_of_accounts(&self, _version: u64) -> Result<u64, Error> {
-        // TODO(joshlind): implement this once DbReaderWriter supports these calls.
-        Err(Error::UnexpectedErrorEncountered(
-            "Unimplemented! This API call needs to be implemented!".into(),
-        ))
-    }
+/// Calculate `(start..=end).len()`. Returns an error if `end < start` or
+/// `end == u64::MAX`.
+fn range_len(start: u64, end: u64) -> Result<u64, Error> {
```

Comment:
Given the amount of ranges & exclusive / inclusive stuff, I'd rename this to `inclusive_range_len` then you don't have to read this comment to get it

Reply:
Renamed it to `inclusive_range_len` (but, I'd also like to leave the comment 😄)